src/FormField.js

```diff
@@ -60,38 +60,43 @@ class InputField extends React.Component {
 
   hasValue = () => {
     const { children } = this.props
-    return React.Children.toArray(children).reduce(
-      (a, child) => a || (child && child.type === Input && !!child.props.value),
-      false
-    )
+    return React.Children
+      .toArray(children)
+      .reduce(
+        (a, child) =>
+          a ||
+          (((child && child.type === Input) || child.type === Select) &&
+            child.props.value),
+        false
+      )
   }
 
   render() {
     const { label, icon, children, onChange, ...props } = this.props
 
-    let InputChild
-    let inputPosition = -1
+    let FieldChild
+    let position = -1
     let LabelChild
     let BeforeIcon
     let AfterIcon
-    let inputId
-    let inputPlaceholder
+    let fieldId
+    let fieldPlaceholder
     let iconAdjustment
 
     React.Children.forEach(children, (child, index) => {
       if (child) {
         if (child.type === Label) {
           LabelChild = child
         }
-        if (child.type === Input) {
-          inputPosition = index
-          InputChild = child
-          inputId = child.props.id
+        if (child.type === Input || child.type === Select) {
```

Comment:
Seems like the `if` condition grows a little bit with the `||` operator. Since there are these `let` variables declared atop, perhaps, the "grown" `if` logic can be "_hoisted_" out to a declared variable along with the ones above?

Reply:
To keep the PR small I followed the existing pattern but in terms of scalability I was thinking of creating an issue to get this component cleaned up.

Reply:
Nice idea @zerodom30 about opening a new issue to discuss and/or ideate about the potential "_cleaning up_" pattern of this component. Let's try out the issue when you have time. See you on the issue. 😸

Reply:
#301 issue has been created and refers to this thread for the thought of having an issue to discuss more simplistic JavaScript pattern to follow up to `if` & inner `if` blocks in to to-be `FormField`. Feel frees to chime in to discuss on that issue.

Reply:
Nice will do.